parsers/ENTSOE.py

```diff
@@ -622,6 +717,35 @@ def fetch_production(zone_key, session=None, target_datetime=None, logger=loggin
     return list(filter(validate_production, data))
 
 
+def fetch_production_per_units(zone_key, session=None, target_datetime=None, logger=logging.getLogger(__name__)):
+    """
+    Returns a list of all production units and production values as a list
+    of dictionaries
+    """
+    if not session:
+        session = requests.session()
+    domain = ENTSOE_EIC_MAPPING[zone_key]
+    data = []
+    # Iterate over all psr types
+    for k in ENTSOE_PARAMETER_DESC.keys():
+        try:
+            values = [ v for v in parse_production_per_units(
+                query_production_per_units(k, domain, session, target_datetime)) or [] if v is not None ]
+            for v in values:
+                if not v: continue
+                v['datetime'] = v['datetime'].datetime
+                v['source'] = 'entsoe.eu'
+                if not v['unitName'] in ENTSOE_UNITS_TO_ZONE:
+                    logger.warning('Unknown unit %s with id %s' % (v['unitName'], v['unitKey']))
+                else:
+                    v['zoneKey'] = ENTSOE_UNITS_TO_ZONE[v['unitName']]
+            if values:
+                data.extend([ v for v in values if v.get('zoneKey', None) == zone_key ])
```

Comment:
could you not remove the values not corresponding to the current `zone_key` while in the loop to avoid another list comprehension?
(and after / before brackets are not `PEP 8`-y)